src/status_im2/contexts/chat/composer/actions/view.cljs

```diff
@@ -147,11 +148,23 @@
                                                   50)}]))
        :max-duration-ms                    constants/audio-max-duration-ms}]]))
 
+(defn request-camera-permission
+  []
+  (rf/dispatch
+   [:request-permissions
+    {:permissions [:camera]
+     :on-allowed  #(rf/dispatch [:navigate-to :camera-screen])
+     :on-denied   #(rf/dispatch
+                    [:toasts/upsert
+                     {:icon           :i/info
+                      :icon-color     colors/danger-50
+                      :override-theme :light
+                      :text           (i18n/label :t/camera-permission-denied)}])}]))
 
 (defn camera-button
   []
   [quo/button
-   {:on-press #(js/alert "to be implemented")
+   {:on-press #(request-camera-permission)
```

Comment:
```suggestion
   {:on-press request-camera-permission
```

Reply:
@smohamedjavid I think something better to do is to create a `device_permissions` file to avoid permissions function duplications. I have created a file at `utils.device_permissions.cljs` to have all of the permissions functions in one file.

Reply:
Yes, it needs some refactoring. For another PR :)